LoopKitUI/Views/QuantityPicker.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  QuantityPicker.swift
+//  LoopKitUI
+//
+//  Created by Michael Pangburn on 4/23/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+import HealthKit
+import LoopKit
+
+
+private struct PickerValueBoundsKey: PreferenceKey {
+    static let defaultValue: [Anchor<CGRect>] = []
+
+    static func reduce(value: inout [Anchor<CGRect>], nextValue: () -> [Anchor<CGRect>]) {
+        value.append(contentsOf: nextValue())
+    }
+}
+
+public struct QuantityPicker: View {
+    @Binding var value: HKQuantity
+    var unit: HKUnit
+    var guardrail: Guardrail<HKQuantity>
+
+    private let possibleValues: [Double]
+    private let formatter: NumberFormatter
+
+    public init(value: Binding<HKQuantity>, unit: HKUnit, stride: HKQuantity, guardrail: Guardrail<HKQuantity>) {
+        self._value = value
+        self.unit = unit
+        self.guardrail = guardrail
+
+        self.possibleValues = guardrail.allValues(stridingBy: stride, unit: unit).map { $0.doubleValue(for: unit) }
```

Comment:
I believe we should change some of the pickers (like isf) to not have a fixed stride and instead have a set of sensible values, with bigger strides at the top end of the range.  But we can probably address that at that point.

Reply:
Makes sense to me. Since the backing storage for the values is just an array, we can provide additional initializers for non-fixed strides or to allow the callee to pass the full array directly.